conans/client/cmd/profile.py

```diff
@@ -19,16 +19,23 @@ def _get_profile_keys(key):
 
     return first_key, rest_key
 
+def _find_profiles(folder, subdirectory_relative_path):
+    profiles = []
+    if os.path.exists(folder):
```

Comment:
I think that the python idiom for traversing children folder is ``os.walk()``, instead of recursing yourself.
Please have a look and try to implement it with ``os.walk()`` and do not hesitate to ask for help if needed. Thanks!